server/pkg/controller/stripe.go

```diff
@@ -341,8 +345,18 @@ func (c *StripeController) handleCustomerSubscriptionUpdated(event stripe.Event)
 	}
 
 	userID := currentSubscription.UserID
-	switch stripeSubscription.Status {
-	case stripe.SubscriptionStatusPastDue:
+	newSubscription, err := c.getEnteSubscriptionFromStripeSubscription(userID, stripeSubscription)
+	if err != nil {
+		return ente.StripeEventLog{}, stacktrace.Propagate(err, "")
+	}
+	if currentSubscription.ProductID == newSubscription.ProductID {
+		// Webhook is reporting an outdated update that was already verified
+		// no-op
+		log.Warn("Webhook is reporting an outdated purchase that was already verified stripeSubscriptionID:", stripeSubscription.ID)
+		return ente.StripeEventLog{UserID: userID, StripeSubscription: stripeSubscription, Event: event}, nil
+	}
+	c.BillingRepo.ReplaceSubscription(currentSubscription.ID, newSubscription)
```

Comment:
this was required to handle the of failed sepa update, where the user's subscription has directly changed to past_due, 

In which our earlier code would not updated the local museum subscription details